### ISSUE TITLE
Almost systematically the orchestration instances were started stuck in the pending state

### ISSUE THREAD

**Issue by AlessandroH** [Azure/azure-functions-durable-extension#1365]
### Description
> We have a durable function that randomly but very often remains pending for several seconds when it is instantiated (even more than 50). Then it suddenly unlocks and runs quickly. Sometimes, however, it does not go into the pending state and execution is very fast

>This is just an example of one of our orchestrations but we have many with the same behavior
 Orchestration Instance ID 281bcdf9-2c15-47e3-8fb6-36041562873c
Region that orchestration executed in Azure WestEurope
Timestamp in UTC of affected orchestration.  2020-05-26T20:33:47.559Z

Just to give another information, we have another subscription with a similar deployment that works perfectly

Thanks in advance, Alessandro

**Comment by cgillum** [comment]
Hi @AlessandroH, thanks for providing these details. I can confirm that it took about 15 seconds for your orchestration to start running. I didn't notice anything abnormal about this specific instance, but I did notice some app-wide issues that *might* explain some of these delays.

One exception I detected that is occurring in the Durable Task Framework:

```
System.ArgumentOutOfRangeException: The argument 'visibilityTimeout' is smaller than minimum of '00:00:00'
Parameter name: visibilityTimeout
   at Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.AssertInBounds[T](String paramName, T val, T min, T max)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageImpl(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageAsync(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options, OperationContext operationContext, CancellationToken cancellationToken)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageAsync(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options, OperationContext operationContext)
   at DurableTask.AzureStorage.Messaging.TaskHubQueue.AbandonMessageAsync(MessageData message, SessionBase session) in C:\source\durabletask\src\DurableTask.AzureStorage\Messaging\TaskHubQueue.cs:line 238
```

This is clearly a bug (integer overflow) and I will make a fix for this. However, I don't think it explains the root cause about why you have messages that repeatedly fail to be processed (most of the failing messages are external events). I'll need to take a closer look.

BTW, if you need a guaranteed resolution to this problem, I suggest opening a support ticket for Azure Functions and referencing this GitHub issue.

**Comment by AlessandroH** [comment]
Hi Chris,
thanks for your support. Our problem is the duration of the Pending state. We have overcome the problem a bit by putting the following settings:
  "extensions": {
    "durableTask": {
      "hubName": "SmartTaskHub",
      "storageProvider": {
        "maxQueuePollingInterval": "00:00:04",
        "partitionCount": 4
      }
    }
}
What do you think about that?
I'll try to follow your suggestion to open a support ticket.
Thanks, Alessandro

**Comment by cgillum** [comment]
Aha. If changing the queue polling interval resolved the issue for you, it means the delay is natural because of how we do exponential backoff polling of the control queues. The maximum delay is ~30 seconds if we go a long time without finding any messages. We do this to avoid incurring excessive storage transaction costs for apps that are mostly idle. The unfortunate side effect is that it can cause long delays in starting new orchestrations for these mostly idle apps. Note that this only applies to apps that are scaled out to multiple VMs, however.

I did not think to look for this when I investigated your app earlier. I checked again just now and I can confirm this is the case. Orchestration **281bcdf9-2c15-47e3-8fb6-36041562873c** was sent to the queue named "xxx-control-02" by "VM1" at 2020-05-26 20:33:47 UTC. "VM2" was assigned to this queue. However, VM2 had not seen any messages in "xxx-control-02" since 2020-05-26 20:32:35 (~72 seconds), which is why it started waiting longer between polls. This explains why it took about 15 seconds before the orchestration start message got picked up.

I think the issue is understood and is (unfortunately) currently by design. The error message I mentioned previously is not related. There is no need to open a support ticket and workaround is correct. Let me know if you have any concerns - otherwise I think we can close this issue.

For reference, here is another issue which tracks the same problem, but before we had the workaround: https://github.com/Azure/azure-functions-durable-extension/issues/606
